bwapi/Library/BWAPILIB/Source/Types/UnitType.cpp

```diff
@@ -554,6 +554,29 @@ namespace BWAPI
     };
   }
 
+  namespace unitSpeedUpgrades
+  {
+      using namespace UpgradeTypes;
+      static const UpgradeType speedUpgrade[UnitTypes::Enum::MAX] = {

```

Comment:
These are off-by-1 after the Vulture(except for Mojo scout), unknown needs to return unknown, and Torrasque/Raynor(vulture) are missing their upgrades. [doc](https://docs.google.com/spreadsheets/d/1rV-Xkradt6CjtVyNMcJYsYsC9KNO_hAC6TxyrF-RrMc/edit?usp=sharing)

This should hopefully be correct:

```
None, None, Ion_Thrusters, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, Ion_Thrusters, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, Metabolic_Boost, Muscular_Augments, Anabolic_Synthesis, None, None, Pneumatized_Carapace, None, None, None, None, None,
Anabolic_Synthesis, None, None, None, None, Muscular_Augments, Metabolic_Boost, None, None, Pneumatized_Carapace, None, None,
None, None, None, None, None, Leg_Enhancements, None, None, None, Gravitic_Drive, Gravitic_Thrusters, None,
None, None, None, None, None, Leg_Enhancements, None, None, Gravitic_Thrusters, None, None, None,
Gravitic_Boosters, None, None, None, Gravitic_Thrusters, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, None, None, None, None, None, None, None,
None, None, None, None, None, Unknown
```